### PR TITLE
Add javascript checkout parameter validation

### PR DESCRIPTION
**Pull request by okenshields**
## Summary

Add javascript checkout parameter validation

## Related issues

Fix #

## Changelog

* added validation and tests to checkout parameters

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -9,7 +9,8 @@
   },
   "main": "index.js",
   "scripts": {
-    "lint": "eslint ./ --ext .js"
+    "lint": "eslint ./ --ext .js",
+    "test": "jest"
```
Comment (hukid) [inline package.json]: Can we add this to travis ci?

File: src/utils.js [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+*/
+
+function hasNonNullProperty(obj, property) {
+  if (Object.prototype.hasOwnProperty.call(obj, property) && typeof obj[property] !== 'undefined' && obj[property] !== null) {
```
Comment (hukid) [inline src/utils.js]: if obj is null or undefined, will this throw?

File: src/utils.js [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+    return true;
+  }
+  return false;
+}
+
+export default function ValidateCheckoutParameters(checkoutParams) {
```
Comment (hukid) [inline src/utils.js]: can checkoutParams be null or undefined?

File: package.json (hunk 1 above)
Comment (okenshields) [reply package.json]: Yes, we should. Added.

File: src/utils.js (hunk 2 above)
Comment (okenshields) [reply src/utils.js]: Yes, added additional validation in calling method ValidateCheckoutParameters

File: src/utils.js (hunk 3 above)
Comment (okenshields) [reply src/utils.js]: Added addition validation for that.

File: src/utils.js [hunk 4]
```diff
@@ -26,6 +26,10 @@ export default function ValidateCheckoutParameters(checkoutParams) {
   paramError.debugCode = 'rn_checkout_invalid_parameter';
   paramError.message = 'Something went wrong. Please contact the developer of this application and provide them with this error code: rn_checkout_invalid_parameter';
   paramError.debugMessage = 'Invalid parameter found in checkout parameters. ';
+  if (checkoutParams === undefined || checkoutParams === null) {
```
Comment (hukid) [inline src/utils.js]: You can simplify it to `if(!checkoutParams)`

**Pull request merged**